dev-tools/mage/manifest/manifest.go

```diff
@@ -68,16 +69,75 @@ func DownloadManifest(manifest string) (tools.Build, error) {
 }
 
 func resolveManifestPackage(project tools.Project, pkg string, reqPackage string, version string) []string {
+	var val tools.Package
+	var ok bool
+
+	// Try the normal/easy case first
 	packageName := fmt.Sprintf("%s-%s-%s", pkg, version, reqPackage)
-	val, ok := project.Packages[packageName]
+	val, ok = project.Packages[packageName]
 	if !ok {
-		return nil
+		// If we didn't find it, it may be an Independent Agent Release, where
+		// the opted-in projects will have a patch version one higher than
+		// the rest of the projects, so we need to seek that out
+		if mg.Verbose() {
+			log.Printf(">>>>>>>>>>> Looking for package [%s] of type [%s]", pkg, reqPackage)
+		}
+
+		var foundIt bool
+		for pkgName := range project.Packages {
+			if strings.HasPrefix(pkgName, pkg) {
+				firstSplit := strings.Split(pkgName, pkg+"-")
+				if len(firstSplit) < 2 {
+					continue
+				}
+
+				secondHalf := firstSplit[1]
+				// Make sure we're finding one w/ the same required package type
+				if strings.Contains(secondHalf, reqPackage) {
+
+					// Split again after the version with the required package string
+					secondSplit := strings.Split(secondHalf, "-"+reqPackage)
+					if len(secondSplit) < 2 {
+						continue
+					}
+
+					// The first element after the split should normally be the version
+					pkgVersion := secondSplit[0]
+					if mg.Verbose() {
+						log.Printf(">>>>>>>>>>> Using derived version for package [%s]: %s ", pkgName, pkgVersion)
+					}
+
+					// Create a project/package key with the package, derived version, and required package
+					foundPkgKey := fmt.Sprintf("%s-%s-%s", pkg, pkgVersion, reqPackage)
+					if mg.Verbose() {
+						log.Printf(">>>>>>>>>>> Looking for project package key: [%s]", foundPkgKey)
+					}
+
+					// Get the package value, if it exists
+					val, ok = project.Packages[foundPkgKey]
+					if !ok {
+						continue
+					}
+
+					if mg.Verbose() {
+						log.Printf(">>>>>>>>>>> Found package key [%s]", foundPkgKey)
+					}
+
+					foundIt = true
+				}
+			}
+		}
+
+		if !foundIt {
+			return nil
+		}
```

Comment:
Should this actually be an error? I am worried about silently dropping a package from the build.

Reply:
Good call, I'll make it an error.

Reply:
I just tested this out, and I don't think it should be an error.  The logic for whether or not something was found is here, and seems to just log that "Project [%s] does not have [%s]".

https://github.com/elastic/elastic-agent/blob/main/dev-tools/mage/manifest/manifest.go#L125-L139

Going to leave it as returning `nil` for now.

I do think there should be some checking at some point to verify all expected packages for a given os/arch exists.  That probably would have caught when endpoint was no longer being packaged.

That said, I'm not sure I understand all of the intricacies to make that change right now.

Reply:
This does need to be an error. We have an internal list of expected binaries, and this ensures we can find all of them in the manifest. One of them missing is a fatal error.

I suspect there are some platform dependencies to handle though.

Reply:
Handled the platform dependencies.